Add option 'enabled'

Koblaid opened the pull request:
PR for #53 

Is a unit test for this required? I was not sure how to write one... ;-)

coryvirok commented on rollbar.js:
```diff
@@ -81,7 +81,7 @@ exports.init = function (accessToken, options) {
    */
   if (!initialized) {
-    if (!accessToken) {
+    if (!accessToken && options.enabled !== false) {
```
I would remove this check in case `options` is not passed into `init()`.

Koblaid replied:
Oh, right, I didn't think of that. The problem would be solved if I move 
```
options = options || {};
```
above the condition.

The reason for this condition is that in development mode I don't have a accessToken, and since its not an error when enabled = false the warning is redundant.

Should I remove it nevertheless? Or just fix the empty-options-case?

coryvirok replied:
I'd move the `options = options || {}` above this check and add in `if (!accessToken && options.enabled)`

coryvirok commented on lib/notifier.js:
```diff
@@ -298,11 +300,18 @@ function addItem(item, callback) {
-      apiClient.postItem(data, function (err, resp) {
-        if (typeof callback === 'function') {
-          callback(err, data, resp);
+      if(SETTINGS.enabled){
```
I would move this check to happen before the call to `buildItemData()` since we are basically neutering the call to `addItem()` if `options.enabled === false`.

Koblaid replied:
Calling buildItemData() even when not actually sending the error has the advantage that all the code parsing and building the error is still run. This way potential errors in Rollbar occur while developing and not only when the app is run in production with Rollbar enabled.

Of course you could argue that setting enabled = false should disable *as much* code as possible in Rollbar, not as little.

Should I still move the check before buildItemData().

coryvirok replied:
I think I prefer to disable as much code as possible which is in line with the change below regarding the access token check. E.g. I could see `buildItem()` expecting an access token and breaking if one isn't provided.

Koblaid commented:
I just changed `addItem()` so that `buildItemData()` is not called when options.enabled = false and moved `options = options || {};` to the top of the function.

Using `if (!accessToken && options.enabled)` in rollbar.init() would not work if the user doesn't set the enabled parameter. The default value for `enabled` is only set in notifier.js and `enabled` would be `undefined` at this moment.

But I don't really like the current solution `options.enabled !== false`. The way the options parsing and setting of default values is done restricts the usage of the option to the file in which the default is defined. I think it would be worthwile to extract the option parsing and the default options into rollbar.js or an extra file. Then the default options could be applied exactly once and the resulting options object could be sent to the submodules via the init-functions. This way every modul could access all options. And the default values, options parsing and warning messages would be in one place. But that should probably be a different issue/pull request.

coryvirok commented on rollbar.js:
```diff
@@ -81,12 +81,12 @@ exports.init = function (accessToken, options) {
   if (!initialized) {
-    if (!accessToken) {
+    options = options || {};
+    if (!accessToken && options.enabled !== false) {
```
Cool. Can you also update the comment above to list `enabled` as an option?

coryvirok commented:
Good point about `options.enabled` not being set and also about moving the options into either rollbar.js or some other top-level module for global access.

razorman8669 commented:
any progress on this issue @coryvirok

coryvirok commented:
Sorry for the slow response to this. Merged and published to NPM. Thanks!
